### PR TITLE
Fix light load bug

### PR DESCRIPTION
**Pull request by Binyang2014**
Fix lightLoadExecutionPlan issue.
An execution context many have multi device execution plans. These plans share the channel connections which are constructed before.
A deviceExecutionPlanKey is introduced to identify these plans. We can get the current device execution plan key via:
`contexts.currentDevicePlan`

### CODE REVIEW

File: src/executor/executor.cc [hunk 1]
```diff
@@ -22,21 +22,62 @@ struct ExecutionContextKey {
            recvBuffSize == other.recvBuffSize && plan == other.plan;
   }
 };
+
+struct DeviceExecutionPlanKey {
+  size_t inputMessageSize;
+  size_t outputMessageSize;
+  size_t contsSrcOffset;
```
Comment (yzygitzh) [inline src/executor/executor.cc:29]: `contsSrcOffset` -> `constSrcOffset`?

File: src/executor/executor.cc [hunk 2]
```diff
@@ -84,15 +126,24 @@ struct Executor::Impl {
                                          size_t outputMessageSize, size_t contsSrcOffset, size_t constDstOffset,
                                          size_t sendBufferSize, size_t recvBufferSize, const ExecutionPlan& plan) {
     ExecutionContextKey key = {sendbuff, recvbuff, sendBufferSize, recvBufferSize, plan.impl_->name};
+    DeviceExecutionPlanKey devicePlanKey = {inputMessageSize, outputMessageSize, contsSrcOffset, constDstOffset};
```
Comment (yzygitzh) [inline src/executor/executor.cc:129]: Same as above

**Review by yzygitzh — CHANGES_REQUESTED** [review]

File: src/executor/executor.cc (hunk 1 above)
Comment (Binyang2014) [reply src/executor/executor.cc]: Fixed

File: src/executor/executor.cc (hunk 2 above)
Comment (Binyang2014) [reply src/executor/executor.cc]: Fixed

**Review by yzygitzh — APPROVED** [review]

**Pull request merged**